profiles/views.py

```diff
@@ -184,15 +190,16 @@ def perform_destroy(self, instance):
         instance.save()
 
 
-class ViewedCompanyList(ListCreateAPIView):
+class ViewedCompanyCreate(CreateAPIView):
     serializer_class = ViewedCompanySerializer
-    permission_classes = (IsAuthenticated,)
-    pagination_class = ForumPagination
-
-    def get_queryset(self):
-        user_id = self.request.user.id
-        return ViewedCompany.objects.filter(user=user_id).order_by(
-            "company_id"
+    queryset = ViewedCompany.objects.all()
+    permission_classes = [
+        (RequestIsCreate & (~IsOwner | ~IsAuthenticated)),
+    ]
+
+    def perform_create(self, serializer):
+        serializer.save(
+            user=self.request.user, company=self.request.data.get("company")
```

Comment:
Don't overuse `self.request.data`. The stuff in `request.data` is handled in a serializer. You shouldn't need to touch this stuff. 

Very similarly with `return Response()`, if you write `relf.request.data.get(blah-blah)` — 99% you're doing something wrong and there is a better way (alongside with some important stuff from DRF to learn).

Reply:
Fixed.